packages/expo-modules-core/android/src/main/java/expo/modules/core/interfaces/ReactActivityHandler.java

```diff
@@ -18,4 +23,27 @@
   default ReactRootView createReactRootView(Activity activity) {
     return null;
   }
+
+  /**
+   * Gives modules a chance to create a ViewGroup that is used as a container for the "real"
+   * ReactRootView, which is added as a child to the container if non-null.
+   * @return a ReactRootView instance to be used as a container, or null if no container is needed
+   */
+  @Nullable
+  default ReactRootView createReactRootViewContainer(Activity activity) {
```

Comment:
is the container necessary to be a `ReactRootView`? it would be better if `ViewGroup` is enough. i am afraid if there are any problems for nested ReactRootView.

Reply:
oh i got the reason, because `createReactRootView` requires a ReactRootView but not a ViewGroup: https://github.com/expo/expo/blob/4714e7a91dd300e6baa695ab89bedafc088d67f6/packages/expo/android/src/main/java/expo/modules/ReactActivityDelegateWrapper.kt#L35

how about we insert the ReactRootView container in `loadApp`?
https://github.com/facebook/react-native/blob/189c2c8958442541c6b4f42860b2943ece612da2/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java#L92

Reply:
cool idea, this seems to work 🎉 I've simply added a second call to `setContentView` which doesn't seem to cause any issues, but let me know if you meant actually overriding and not calling the delegate `loadApp` at all.

Reply:
cool idea. this works 🎉